networkx/algorithms/approximation/steinertree.py

```diff
@@ -46,22 +57,172 @@ def metric_closure(G, weight="weight"):
     return M
 
 
+def mehlhorn_steiner_tree(G, S, weight):
```

Comment:
Since these helper functions aren't exposed publicly in `__all__`, their name should start with an underscore (by convention in NetworkX) to indicate that they are private functions to this module.